src/app/app.js

```diff
@@ -20,36 +12,65 @@ angular
     'ui.grid',
     'userMessage',
     'roleProxy',
-    'ui.bootstrap.datepicker'
+    'ui.bootstrap.datepicker',
+    'oauthRequest',
   ])
   .config(function ($routeProvider) {
     $routeProvider
       .when('/', {
-        templateUrl: 'views/main.html',
-        controller: 'MainCtrl'
+        templateUrl: 'app/views/main.html',
+        controller: 'MainController'
       })
       .when('/login', {
-        templateUrl: 'views/login.html',
+        templateUrl: 'app/views/login.html',
         controller: 'LoginController'
       })
       .when('/users', {
-        templateUrl: 'views/users/list.html',
+        templateUrl: 'app/views/users/list.html',
         controller: 'UserListController'
       })
       .when('/users/new', {
-        templateUrl: 'views/users/create.html',
+        templateUrl: 'app/views/users/create.html',
         controller: 'UserCreateController'
       })
       .when('/users/:userId', {
-        templateUrl: 'views/users/edit.html',
+        templateUrl: 'app/views/users/edit.html',
         controller: 'UserEditController'
       })
-      .when('/datacollection/workday', {
-        templateUrl: 'views/dataCollection/workday.html',
-        controller: 'WorkdayController'
+      .when('/workday/create', {
+        templateUrl: 'app/views/workday/create.html',
+        controller: 'WorkdayCreateController'
+      })
+      .when('/worktypes', {
+        templateUrl: 'app/views/worktype/list.html',
+        controller: 'WorktypeListController'
+      })
+      .when('/worktypes/new', {
+        templateUrl: 'app/views/worktype/create.html',
+        controller: 'WorktypeCreateController'
+      })
+      .when('/worktypes/:id', {
+        templateUrl: 'app/views/worktype/edit.html',
+        controller: 'WorktypeEditController'
+      })
+      .when('/locations', {
+        templateUrl: 'app/views/location/list.html',
+        controller: 'LocationListController'
+      })
+      .when('/locations/new', {
+        templateUrl: 'app/views/location/create.html',
+        controller: 'LocationCreateController'
+      })
+      .when('/locations/:id', {
+        templateUrl: 'app/views/location/edit.html',
+        controller: 'LocationEditController'
+      })
+      .when('/logout', {
+        templateUrl: 'app/views/main.html',
+        controller: 'LogoutController'
```

Comment:
I would like to find a way for the paths to be created by convention, automatically, instead of through configuration. Although it really isn't that onerous to add a new set of three lines for every page in the application.